xbmc/music/MusicDatabase.cpp

```diff
@@ -3401,7 +3401,12 @@ bool CMusicDatabase::GetArtistsNav(const std::string& strBaseDir, CFileItemList&
     else if (idSong > 0)
       musicUrl.AddOption("songid", idSong);
 
-    musicUrl.AddOption("albumartistsonly", albumArtistsOnly);
+    // Override albumArtistsOnly parameter (usually externally set to SETTING_MUSICLIBRARY_SHOWCOMPILATIONARTISTS)
+    // when local option already present in muscic URL thus allowing it to be an option in custom nodes
+    const CUrlOptions::UrlOptions& options = musicUrl.GetOptions();
```

Comment:
Why not use `musicUrl.HasOption("albumartistsonly")` instead of all the iterator stuff?